### PR TITLE
OP-1192 | Add new inventory for all Medicals or for specific Medical

### PR DESCRIPTION
**Pull request by JantBogard**
See [OP-1192](https://openhospital.atlassian.net/browse/OP-1192)

[OP-1192]: https://openhospital.atlassian.net/browse/OP-1192?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java [hunk 1]
```diff
@@ -330,21 +533,45 @@ class InventoryRowModel extends DefaultTableModel {
...
 
+        public InventoryRowModel() {
+            if (allRadio.isSelected()) {
+                try {
+                    inventoryRowList = loadNewInventoryTable(null);
+                } catch (OHServiceException e) {
+                    inventoryRowList = new ArrayList<>();
+                    OHServiceExceptionUtil.showMessages(e);
+                }
+            } else if (specificRadio.isSelected() && code != null && !code.trim().equals("")) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java:544]: If you think the `code` can have trailing spaces then why do you use it in line 546 with those trailing spaces present?

File: bundle/language_en.properties [hunk 2]
```diff
@@ -665,14 +665,23 @@ angal.hospital.visitendhour.txt
 angal.hospital.visitstarthour.txt                                                                      = Visit start hour
 angal.lab.allnegative.txt                                                                              = All negative.
 angal.inventory.allproduct.radio                                                                       = All products
+angal.inventory.anotherinventoryonthiswardstillanotheroneisinprogress                                  = Another inventory on this ward still another one is in progress
```
Comment (dbmalkovsky) [inline bundle/language_en.properties:668]: First the suffix on this entry should be ".msg".   Second, the text of the message isn't very clear; actually it doesn't make any sense.

File: src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java (hunk 1 above)
Comment (JantBogard) [reply src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java:544]: Please, I don't quite understand your question

Comment (dbmalkovsky) [reply src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java:544]: You trim the `code` field to test it:   `!code.trim().equals("")`.   This indicates to me that you expect trailing spaces could be present.

Then you use it in line 546 without trimming:  `loadNewInventoryTable(code);`  Here spaces could or could not be present.  One would not expect that the number of trailing spaces would be significant.

Search for `trim()` in this PR and you will see other occasions of trimming the value before using it in a method call.

Comment (JantBogard) [reply src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java:544]: ok

File: bundle/language_en.properties [hunk 3]
```diff
@@ -665,14 +665,23 @@ angal.hospital.visitendhour.txt
 angal.hospital.visitstarthour.txt                                                                      = Visit start hour
 angal.lab.allnegative.txt                                                                              = All negative.
 angal.inventory.allproduct.radio                                                                       = All products
+angal.inventory.cannotcreateanotherinventorywithotherinprogressinthisward                              = cannot create an other inventory with other in progress in this ward.
```
Comment (mwithi) [inline bundle/language_en.properties:668]:
```suggestion
angal.inventory.cannotcreateanotherinventorywithotherinprogressinthisward.msg                              = Cannot create an other inventory with other in progress in this ward.
```

File: bundle/language_en.properties [hunk 4]
```diff
@@ -681,12 +690,17 @@ angal.inventory.state.label
 angal.inventory.state.canceled.txt                                                                     = Canceled
 angal.inventory.state.inprogress.txt                                                                   = In progress
 angal.inventory.state.validate.txt                                                                     = Validate
-angal.inventory.theorticalqty.col                                                                      = Theoretical quantity
+angal.inventory.theorticalqty.col                                                                      = Theoretical
+angal.inventory.totalprice.col                                                                             = Total price quantity
+angal.inventory.unitprice.col                                                                          = Unit price
```
Comment (mwithi) [inline bundle/language_en.properties:695]:
```suggestion
angal.inventory.theorticalqty.col                                                                      = Theoretical
angal.inventory.totalprice.col                                                                         = Total price quantity
angal.inventory.unitprice.col                                                                          = Unit price
```

File: src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java [hunk 5]
```diff
@@ -330,21 +542,51 @@ class InventoryRowModel extends DefaultTableModel {
...
+            } else if (c == 7) {
                 return Double.class;
             }
             return null;
         }
 
         public int getRowCount() {
-            return 0;
+            if (inventoryRowSearchList == null)
+                return 0;
```
Comment (mwithi) [inline src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java:588]:
```suggestion
            if (inventoryRowSearchList == null) {
                return 0;
            }
```

File: bundle/language_en.properties
Comment (mwithi) [inline bundle/language_en.properties:1]: Are `angal.menu.inventoryward` and `angal.menu.btn.inventoryward` missing?

File: src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java [hunk 6]
```diff
@@ -89,33 +144,69 @@ public static void addInventoryListener(InventoryListener l) {
     private JButton validateButton;
     private JScrollPane scrollPaneInventory;
     private JTable jTableInventoryRow;
+    private List<MedicalInventoryRow> inventoryRowList;
+    private List<MedicalInventoryRow> inventoryRowSearchList;
     private String[] pColums = { MessageBundle.getMessage("angal.common.code.txt"),
             MessageBundle.getMessage("angal.inventory.medical.col"),
+            MessageBundle.getMessage("angal.inventory.lotcode.col"),
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java:151]: Please review code in other table definitions; column titles should be in uppercase; use `toUpperCase()`

File: bundle/language_en.properties
Comment (JantBogard) [reply bundle/language_en.properties:1]: No.

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**